src/hotspot/share/memory/arena.hpp

```diff
@@ -81,19 +85,29 @@ class Chunk {
   char* bottom() const          { return ((char*) this) + aligned_overhead_size();  }
   char* top()    const          { return bottom() + _len; }
   bool contains(char* p) const  { return bottom() <= p && p <= top(); }
+
+  void set_stamp(uint64_t v) { _stamp = v; }
+  uint64_t stamp() const     { return _stamp; }
 };
 
+// Arena types (for Compilation Memory Statistic)
 #define DO_ARENA_TAG(FN) \
-  FN(other, Others, Other arenas) \
-  FN(ra, RA, Resource areas) \
-  FN(ha, HA, Handle area) \
-  FN(node, NA, Node arena) \
+  FN(ra,          Resource areas) \
+  FN(node,        C2 Node arena) \
+  FN(comp,        C2 Compile arena) \
+  FN(type,        C2 Type arena) \
+  FN(index,       C2 Index arena) \
```

Comment:
`tag_index` is not used and can be removed, it seems to be subsumed by `tag_reglive`.